### DISCUSSION TITLE
[Guided tours] Issue with Subform Field Highlighting During Guided Tour Creation

### DISCUSSION THREAD

**Discussion by stevenoyer** [joomla/joomla-cms#47062] [2. Cold Features]
### Steps to reproduce the issue

We are currently experiencing an issue when creating a guided tour. In an article, we use a custom field of type "subform" which contains multiple fields. When the guided tour is initiated, we proceed through the steps until we reach the "subform" field. Upon clicking the plus (+) button, the subform field block is created correctly. However, the fields within this block are not highlighted during the guided tour, resulting in the user not receiving the necessary guidance.

However, if we save our article at this point after adding a subform using the plus (+) button and then relaunch the guided tour, the fields are properly highlighted. Therefore, we assume that this issue stems from the fact that the JavaScript script does not take into account the HTML code of the new block, preventing it from targeting the subform fields correctly.

### System information (as much as possible)
Joomla 4.3.4

Thank you for your assistance.

**Comment by obuisard** [comment]
Hello Steven @stevenoyer,
the guided tours do not take into account elements created 'on the fly' in the interface, only elements that exist at the time a step is shown.
I don't have a solution at this time, but we will take that issue into account and see if this is something that can be improved.
Thank you!
